src/adapter/philips_hue/api/mod.rs

```diff
@@ -0,0 +1,234 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#![allow(dead_code)]
+
+extern crate serde_json;
+
+pub mod structs;
+
+use adapter::philips_hue::http;
+use std;
+use std::collections::BTreeMap;
+use std::error::Error;
+use std::time::Duration;
+use std::thread;
+use std::hash::{Hash, SipHasher, Hasher};
+
+
+#[derive(Debug, Hash)]
+struct HueApiToken {
+    iv: String,
+    id: String,
+}
+
+impl HueApiToken {
+    // This implementation offers 64 bit
+    fn as_hash(&self) -> String {
+        let mut s = SipHasher::new();
+        self.hash(&mut s);
+        format!("{:016x}", s.finish()).to_owned()
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct HueApi {
```

Comment:
Refactored to *HueHubApi*.

In the code I very much prefer *Hub* over *Bridge* for brevity.